ocaml/xapi/xapi_observer.ml

```diff
@@ -110,14 +110,139 @@ module Observer : ObserverInterface = struct
     Tracing.Export.Destination.File.set_compress_tracing_files enabled
 end
 
-let supported_components = ["xapi"; "xenopsd"]
+module ObserverConfig = struct
+  type t = {
+      otel_service_name: string
+    ; otel_resource_attributes: (string * string) list
+    ; otel_exporter_zipkin_endpoints: string list
+    ; xs_exporter_bugtool_endpoint: string option
+  }
+
+  let filtered_endpoints regex endpoints =
+    List.filter (fun endpoint -> Str.string_match regex endpoint 0) endpoints
+
+  let config_of_observer ~__context ~component ~observer =
+    let endpoints = Db.Observer.get_endpoints ~__context ~self:observer in
+    let zipkin_reg = Str.regexp "http://.*:9411" in
+    let bugtool_reg = Str.regexp "bugtool$" in
+    {
+      otel_service_name= component
+    ; otel_resource_attributes=
+        Db.Observer.get_attributes ~__context ~self:observer
+    ; otel_exporter_zipkin_endpoints= endpoints |> filtered_endpoints zipkin_reg
+    ; xs_exporter_bugtool_endpoint=
+        ( match endpoints |> filtered_endpoints bugtool_reg with
+        | x :: _ ->
+            Some x
+        | [] ->
+            None
+        )
+    }
+end
+
+module type COMPONENT = sig
+  val component_name : string
+end
+
+module Dom0ObserverConfig (Component : COMPONENT) : ObserverInterface = struct
+  let config_root = "/etc/xensource/observer/"
+
+  let env_vars_of_config (config : ObserverConfig.t) =
+    [
+      "OTEL_SERVICE_NAME=" ^ config.otel_service_name
+    ; "OTEL_RESOURCE_ATTRIBUTES="
+      ^ String.concat ","
+          (List.map (fun (k, v) -> k ^ "=" ^ v) config.otel_resource_attributes)
```

Comment:
using `=` here seems fragile since it's also used for separating the env vars with their values, is this the format that opentelemetry standard dictates?